server/src/main/java/io/spine/server/entity/EntityLifecycle.java

```diff
@@ -316,6 +318,21 @@ public final void onStateChanged(EntityRecordChange change,
         postIfRestored(change, messageIds);
     }
 
+    /**
+     * Posts the {@link MigrationApplied} event.
+     *
+     * @return the event or an empty {@code Optional} if the posting was blocked by the
+     *         {@link #eventFilter}
+     */
+    public final Optional<Event> onMigrationApplied() {
```

Comment:
That's a bit strange that we're able to filter out the migration event, but on the other hand, in the `Migration` we throw an `IllegalStateException` in case of such filtering.